dmoj/cptbox/helper.cpp

```diff
@@ -175,6 +225,26 @@ int cptbox_child_run(const struct child_config *config) {
 #endif
 }
 
+int get_landlock_version() {
+#if !PTBOX_FREEBSD
+    char *sandbox_mode = getenv("DMOJ_SANDBOX_MODE");
+    if (sandbox_mode != nullptr && strcmp(sandbox_mode, "ptrace+seccomp") == 0) {
+        // Allow disabling of landlock.
+        return 0;
+    }
+
+    int landlock_version = landlock_create_ruleset(NULL, 0, LANDLOCK_CREATE_RULESET_VERSION);
+    if (landlock_version >= 0) {
+        return landlock_version;
+    }
+    if (errno == ENOSYS || errno == EOPNOTSUPP)
+        return 0;
+    return -1;
+#else
+    return 0;  // FreeBSD does not have landlock
```

Comment:
We should `abort()` here.

Reply:
How? Consumers of this function throw a python error, but what would be appropriate to do here?

Reply:
Got it. I think `has_landlock` should be implemented here rather than in Python, so we can return false order FREEBSD.